### DISCUSSION TITLE
Use gym without provisioning profile set in xcode

### DISCUSSION THREAD

**Discussion by FFDeveloper** [fastlane/fastlane#18426] [Q&A] [answered]
Hi, is it possibile to use gym to archive and export an ipa providing provisioning profile, code signing identitiy and so on as options without specifying a provisioning profile in xcode?
I'm trying to build with gym from shell a third party app and i know which provisioning profile and code signing identity should I use for each schema/target. The developers should continue to leave the provisioning profile for the target as none on Xcode and I want to force the correct provisioning profile with gym options. I am using --export_options with a correct options plist file but I'm getting the following error:

> $ fastlane gym --scheme preIntSandbox --export_method ad-hoc --export_options ExportOptions_adhoc.plist
> [✔] 🚀
> [18:01:26]: fastlane detected a Gemfile in the current directory
> [18:01:26]: However, it seems like you didn't use `bundle exec`
> [18:01:26]: To launch fastlane faster, please use
> [18:01:26]:
> [18:01:26]: $ bundle exec fastlane gym --scheme preIntSandbox --export_method ad-hoc --export_options ExportOptions_adhoc.plist
> [18:01:26]:
> [18:01:26]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
> [18:01:30]: Resolving Swift Package Manager dependencies...
> [18:01:30]: $ xcodebuild -resolvePackageDependencies -scheme preIntSandbox -project ./********.xcodeproj
> [18:01:30]: ▸ Command line invocation:
> [18:01:30]: ▸     /Applications/Xcode12.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -scheme preIntSandbox -project ./********.xcodeproj
> [18:01:31]: ▸ Resolve Package Graph
> [18:01:33]: ▸ Resolved source packages:
> [18:01:33]: ▸   ...
> [18:01:33]: ▸   ...
> [18:01:33]: ▸ resolved source packages: *****
> [18:01:33]: $ xcodebuild -showBuildSettings -scheme preIntSandbox -project ./********.xcodeproj
> `<PBXResourcesBuildPhase UUID=`688C067223CCA81600808B68`>` attempted to initialize an object with an unknown UUID. `686F5D2B24080E3600D767D0` for attribute: `files`. This can be the result of a merge and the unknown UUID is being discarded.
> [18:01:34]: Detected provisioning profile mapping: {:"it.foo.bar.mobile.myapp"=>"match AdHoc it.foo.bar.mobile.myapp"}
> `<PBXResourcesBuildPhase UUID=`688C067223CCA81600808B68`>` attempted to initialize an object with an unknown UUID. `686F5D2B24080E3600D767D0` for attribute: `files`. This can be the result of a merge and the unknown UUID is being discarded.
> 
> +---------------------------------------------------------------------------+-------------------------------------------------------+
> |                                                      Summary for gym 2.178.0                                                      |
> +---------------------------------------------------------------------------+-------------------------------------------------------+
> | scheme                                                                    | preIntSandbox                                         |
> | export_method                                                             | ad-hoc                                                |
> | export_options.compileBitcode                                             | true                                                  |
> | export_options.destination                                                | export                                                |
> | export_options.method                                                     | ad-hoc                                                |
> | export_options.provisioningProfiles.it.foo.bar.mobile.myapp 				| match AdHoc it.foo.bar.mobile.myapp    				|
> | export_options.signingCertificate                                         | Apple Distribution                                    |
> | export_options.signingStyle                                               | manual                                                |
> | export_options.stripSwiftSymbols                                          | true                                                  |
> | export_options.teamID                                                     | **********                                            |
> | export_options.thinning                                                   | <none>                                                |
> | project                                                                   | ./***********.xcodeproj                               |
> | clean                                                                     | false                                                 |
> | output_directory                                                          | .                                                     |
> | output_name                                                               | preIntSandbox                                         |
> | silent                                                                    | false                                                 |
> | skip_package_ipa                                                          | false                                                 |
> | skip_package_pkg                                                          | false                                                 |
> | build_path                                                                | /Users/***/Library/Developer/Xcode/Archives/2021-03-18|
> | result_bundle                                                             | false                                                 |
> | buildlog_path                                                             | ~/Library/Logs/gym                                    |
> | destination                                                               | generic/platform=iOS                                  |
> | skip_profile_detection                                                    | false                                                 |
> | skip_package_dependencies_resolution                                      | false                                                 |
> | disable_package_automatic_updates                                         | false                                                 |
> | use_system_scm                                                            | false                                                 |
> | xcode_path                                                                | /Applications/Xcode12.app                             |
> +---------------------------------------------------------------------------+-------------------------------------------------------+
> 
> [18:01:34]: $ set -o pipefail && xcodebuild -scheme preIntSandbox -project ./********.xcodeproj -destination 'generic/platform=iOS' -archivePath /Users/***/Library/Developer/Xcode/Archives/2021-03-18/preIntSandbox\ 2021-03-18\ 18.01.34.xcarchive archive | tee /Users/***/Library/Logs/gym/preIntSandbox-preIntSandbox.log | xcpretty
> [18:01:39]: ▸     The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target '****' from project '****')
> [18:01:39]: ▸ ❌  error: "preIntSandbox" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'preIntSandbox' from project '********')
> [18:01:39]: ▸     The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target '****' from project '****')
> [18:01:39]: ▸ ** ARCHIVE FAILED **
>     The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target '****' from project '****')
> 
> ❌  error: "preIntSandbox" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'preIntSandbox' from project '********')
> 
> 
>     The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target '****' from project '****')
> ** ARCHIVE FAILED **
> [18:01:39]: Exit status: 65
> [18:01:39]:
> [18:01:39]: Maybe the error shown is caused by using the wrong version of Xcode
> [18:01:39]: Found multiple versions of Xcode in '/Applications/'
> [18:01:39]: Make sure you selected the right version for your project
> [18:01:39]: This build process was executed using '/Applications/Xcode12.app'
> [18:01:39]: If you want to update your Xcode path, either
> [18:01:39]:
> [18:01:39]: - Specify the Xcode version in your Fastfile
> [18:01:39]: ▸ xcversion(version: "8.1") # Selects Xcode 8.1.0
> [18:01:39]:
> [18:01:39]: - Specify an absolute path to your Xcode installation in your Fastfile
> [18:01:39]: ▸ xcode_select "/Applications/Xcode8.app"
> [18:01:39]:
> [18:01:39]: - Manually update the path using
> [18:01:39]: ▸ sudo xcode-select -s /Applications/Xcode.app
> [18:01:39]:
> 
> +---------------+---------------------------+
> |             Build environment             |
> +---------------+---------------------------+
> | xcode_path    | /Applications/Xcode12.app |
> | gym_version   | 2.178.0                   |
> | export_method | ad-hoc                    |
> | sdk           | iPhoneOS14.0.sdk          |
> +---------------+---------------------------+
> 
> [18:01:39]: ▸ note: Planning build
> [18:01:39]: ▸ note: Constructing build description
> [18:01:39]: ▸ warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target '****' from project '****')
> [18:01:39]: ▸ error: "preIntSandbox" requires a provisioning profile. Select a provisioning profile in the Signing & Capabilities editor. (in target 'preIntSandbox' from project '********')
> [18:01:39]: ▸ warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.0.99. (in target '****' from project '****')
> [18:01:39]:
> [18:01:39]: ⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error
> [18:01:39]: 📋  For the complete and more detailed error log, check the full log at:
> [18:01:39]: 📋  /Users/***/Library/Logs/gym/preIntSandbox-preIntSandbox.log
> [18:01:39]:
> [18:01:39]: Looks like fastlane ran into a build/archive error with your project
> [18:01:39]: It's hard to tell what's causing the error, so we wrote some guides on how
> [18:01:39]: to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
> [18:01:39]: Before submitting an issue on GitHub, please follow the guide above and make
> [18:01:39]: sure your project is set up correctly.
> [18:01:39]: fastlane uses `xcodebuild` commands to generate your binary, you can see the
> [18:01:39]: the full commands printed out in yellow in the above log.
> [18:01:39]: Make sure to inspect the output above, as usually you'll find more error information there
> [18:01:39]:
> 
> [!] Error building the application - see the log above

**Comment by kambala-decapitator** [comment]
try passing `CODE_SIGNING_ALLOWED=NO` build setting and pass `signingCertificate` + `provisioningProfiles` export settings

**Reply by FFDeveloper** [reply]
Thanks but the fix, for me, was to add the --archive_path option

**Comment by FFDeveloper** [comment, marked as answer]
SOLVED: It seems that the issue is solved by adding the --archive_path option

**Reply by Blackclaws** [reply]
Add it to where exactly?
